From a045255d166ab7521355ae1ce041960e7cf9d487 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Wed, 5 May 2021 01:32:04 -0700
Subject: [PATCH] Fix build with gcc-11

* fixes:
/jenkins/mjansa/build/ros/webos-foxy-honister/tmp-glibc/work/qemux86-webos-linux/rplidar-ros/2.0.1-1-r0/git/sdk/src/arch/linux/net_socket.cpp: In member function 'virtual u_result rp::net::SocketAddress::getAddressAsString(char*, size_t) const':
/jenkins/mjansa/build/ros/webos-foxy-honister/tmp-glibc/work/qemux86-webos-linux/rplidar-ros/2.0.1-1-r0/git/sdk/src/arch/linux/net_socket.cpp:170:15: error: ordered comparison of pointer with integer zero ('const char*' and 'int')
  170 |     return ans<=0?RESULT_OPERATION_FAIL:RESULT_OK;
      |            ~~~^~~

inet_ntop:
RETURN VALUE
       On success, inet_ntop() returns a non-null pointer to dst.  NULL is returned if there was an error, with errno set to indicate the error.

Upstream-Status: Pending
Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 sdk/src/arch/linux/net_socket.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sdk/src/arch/linux/net_socket.cpp b/sdk/src/arch/linux/net_socket.cpp
index 0124a13..8357326 100644
--- a/sdk/src/arch/linux/net_socket.cpp
+++ b/sdk/src/arch/linux/net_socket.cpp
@@ -167,7 +167,7 @@ u_result SocketAddress::getAddressAsString(char * buffer, size_t buffersize) con
 
         break;
     }
-    return ans<=0?RESULT_OPERATION_FAIL:RESULT_OK;
+    return ans == NULL ? RESULT_OPERATION_FAIL:RESULT_OK;
 }
 
 
